test/xref2/gen.ml

```diff
@@ -0,0 +1,31 @@
+(** Generates man pages for a set of odocl files and cats them to stdout using [man]. *)
+
+let run fmt =
+  Printf.ksprintf
+    (fun cmd ->
+      let r = Sys.command cmd in
+      if r <> 0 then (
+        Printf.eprintf "Command failed (%d): %s\n" r cmd;
+        exit 1 ))
+    fmt
+
+let rec list_dir_rec acc path =
+  Sys.readdir path
+  |> Array.fold_left
+       (fun acc name ->
+         let p = Filename.concat path name in
+         if Sys.is_directory p then list_dir_rec acc p else p :: acc)
+       acc
+  |> List.sort String.compare
+
+let out_dir = "_out"
+
+let generate input =
+  run "odoc man-generate -o %S %s" out_dir (Filename.quote input)
+
+let () =
+  for i = 1 to Array.length Sys.argv - 1 do
+    generate Sys.argv.(i)
+  done;
+  list_dir_rec [] out_dir
+  |> List.iter (fun p -> run "COLUMNS=80 man -l %s | col -b" p)
```

Comment:
These tests may be very fragile because they depend on `man` and `col`, which are very complicated programs. I'm replacing that with Mdx tests instead.